packages/fx-core/src/core/middleware/envInfoLoader.ts

```diff
@@ -68,6 +68,7 @@ export function EnvInfoLoaderMW(
         targetEnvName = await askTargetEnvironment(ctx, inputs, allowCreateNewEnv, lastUsedEnvName);
       }
       lastUsedEnvName = targetEnvName ?? lastUsedEnvName;
+      ctx.projectSettings.activeEnvironment = lastUsedEnvName;
```

Comment:
Let's confirm with UX designer that whether we should update the activate environment with the last used environment name.

Reply:
Changed to new modal confirm.